rescript/orient.py

```diff
@@ -1,55 +1,68 @@
 # ----------------------------------------------------------------------------
-# Copyright (c) 2021, QIIME 2 development team.
+# Copyright (c) 2022-2022, QIIME 2 development team.
 #
 # Distributed under the terms of the Modified BSD License.
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
-
 import tempfile
 from q2_types.feature_data import DNAFASTAFormat, DNAIterator
 
 from ._utilities import run_command
 
 
 def orient_seqs(sequences: DNAFASTAFormat,
-                reference_sequences: DNAFASTAFormat,
+                reference_sequences: DNAFASTAFormat = None,
                 perc_identity: float = 0.9,
                 query_cov: float = 0.9,
                 threads: int = 1,
                 left_justify: bool = False,
                 ) -> (DNAFASTAFormat, DNAFASTAFormat):
     matched_temp, notmatched = DNAFASTAFormat(), DNAFASTAFormat()
-    # use vsearch to search query seqs against reference database
-    # report orientation of query seqs relative to reference seqs.
-    with tempfile.NamedTemporaryFile() as out:
-        # note: qmask is disabled as DNAFASTAFormat requires all output seqs
-        # to be uppercase. Could loop through output seqs to convert to upper
-        # but which is faster: disabling masking or looping through with skbio?
-        cmd = ['vsearch', '--usearch_global', str(sequences),
-               '--matched', str(matched_temp), '--notmatched', str(notmatched),
-               '--db', str(reference_sequences), '--id', str(perc_identity),
-               '--maxaccepts', '1', '--strand', 'both', '--qmask', 'none',
-               '--query_cov', str(query_cov), '--threads', str(threads),
-               '--userfields', 'qstrand', '--userout', out.name]
-        if left_justify:
-            cmd.append('--leftjust')
-        run_command(cmd)
-        with open(out.name, 'r') as orient:
-            orientations = [line.strip() for line in orient]
+    if (reference_sequences is not None):
+        # use vsearch to search query seqs against reference database
+        # report orientation of query seqs relative to reference seqs.
+        with tempfile.NamedTemporaryFile() as out:
+            # note: qmask is disabled as DNAFASTAFormat requires all output
+            # seqs to be uppercase. Could loop through output seqs to convert
+            # to upper but which is faster: disabling masking or looping
+            # through with skbio?
+            cmd = ['vsearch',
+                   '--usearch_global', str(sequences),
+                   '--matched', str(matched_temp),
+                   '--notmatched', str(notmatched),
```

Comment:
what do you think of raising a warning message or Note that these various parameters are ignored when running in reverse complement mode? (this info can also just be described in the parameter/action descriptions)

Reply:
I tried that in Sidle, and the warning messages weren't raised unless you were in verbose mode. See this forum conversation with Matt:

https://forum.qiime2.org/t/warning-not-raised/21712/

The challenge with adding it to the docs is that right now, RESCRIPt sets a global variable for vsearch parameters that's used for parameters here. I tried to add an update to the docs, let me know if you that thats sufficient or if an expanded doc version needs to be built that adds a warning string to the parameters.

Reply:
Good points, and the changes you already made to the input descriptions do most of what's needed.

Want to just add an extra 1-2 sentences to the action description in `plugin_setup.py`? Something like this (added to the end of the description):

> Alternatively, if no reference sequences are provided as input, all input sequences will be reverse-complemented. In this case, no alignment is performed, and all alignment parameters (`perc_identity`, `query_cov`, `threads`, and `left_justify`) are ignored.